sbnci/Modules/RecoEff/RecoEff_module.cc

```diff
@@ -0,0 +1,417 @@
+	float length = TrackLength(track);
+
+	if(n_tracks_map[trackID] == 0){
+	  track_comp_map[trackID] = comp;
+	  track_pur_map[trackID] = pur;
+	  track_length_map[trackID] = length;
+	}
+	else if(comp > track_comp_map[trackID]) {
+	  track_comp_map[trackID] = comp;
+	  track_pur_map[trackID] = pur;
+	  track_length_map[trackID] = length;
+	}
+
+	n_tracks_map[trackID]++;
+      }
+      else if(daughter->PdgCode() == 11) {
+	const std::vector<art::Ptr<recob::Shower> > showersVec = pfpShowerAssn.at(daughter.key());
+	if(showersVec.size() != 1) continue;
+	const art::Ptr<recob::Shower> shower = showersVec[0];
+
+	float xOutEdge = 175, xInEdge = 1.5, yEdge = 175, zFrontEdge = 30, zBackEdge = 450;
+	float x = shower->ShowerStart().X(), y = shower->ShowerStart().Y(), 
+	  z = shower->ShowerStart().Z();
+
+	if(TMath::Abs(x) > xOutEdge || TMath::Abs(x) < xInEdge ||
+	   TMath::Abs(y) > yEdge || z < zFrontEdge || z > zBackEdge) continue;
+
+	std::vector<art::Ptr<recob::Hit> > showerHits = showerHitAssn.at(shower.key());
+	int trackID = TruthMatchUtils::TrueParticleIDFromTotalRecoHits(clockData,showerHits,true);
+	float comp = Completeness(showerHits,trackID);
+	float pur = Purity(showerHits,trackID);
+
+	std::vector<double> dEdxVec = shower->dEdx();
+	int best_plane = shower->best_plane();
+	float dEdx = -999;
+	if(dEdxVec.size() != 0) dEdx = dEdxVec[best_plane];
+
+	if(n_showers_map[trackID] == 0){
+	  shower_comp_map[trackID] = comp;
+	  shower_pur_map[trackID] = pur;
+	  shower_dEdx_map[trackID] = dEdx;
+	}
+	else if(comp > shower_comp_map[trackID]) {
+	  shower_comp_map[trackID] = comp;
+	  shower_pur_map[trackID] = pur;
+	  shower_dEdx_map[trackID] = dEdx;
+	}
+
+	n_showers_map[trackID]++;
+      }
+    }
+  }
+}
+
+void RecoEff::TruthProcessor(art::Event const &e)
+{
+  art::FindManyP<simb::MCParticle> nuParticleAssn(eHandleNeutrinos,e,fLArGeantModuleLabel);
+
+  for(unsigned int nu_i = 0; nu_i < eHandleNeutrinos->size(); ++nu_i){
+    const art::Ptr<simb::MCTruth> truthNeutrino(eHandleNeutrinos,nu_i);
+    if(truthNeutrino.isNull()) continue;
+    std::vector<art::Ptr<simb::MCParticle> > particles = nuParticleAssn.at(truthNeutrino.key());
+
+    for(auto particle : particles){
+      if(particle->Mother() != 0 || particle->StatusCode() != 1) continue;
+
+      mc_trackID = -999; mc_PDG = -999; mc_x0 = -999; mc_y0 = -999; mc_z0 = -999;
+      mc_xEnd = -999; mc_yEnd = -999; mc_zEnd = -999; mc_pX0 = -999; mc_pY0 = -999;
+      mc_pZ0 = -999; mc_energy0 = -999; mc_momentum = -999; mc_pXEnd = -999; mc_pYEnd = -999;
+      mc_pZEnd = -999; mc_energyEnd = -999; mc_mass = -999; mc_theta_xy = -999; mc_theta_yz = -999;
+      mc_theta_xz = -999; mc_length = -999; reco_nTracks = -999; reco_nShowers = -999; 
+      reco_shower_purity = -999; reco_shower_completeness = -999; reco_track_purity = -999; 
+      reco_track_completeness = -999; reco_track_length = -999; reco_shower_dEdx = -999; 
+      reco_isReconstructed = false;
```

Comment:
This could be put into a Reset function.
Also, you're setting some floating point variables to integer values.
Maybe also define the default value as a constexpr somewhere at the top of the file.

Reply:
Done [here](https://github.com/SBNSoftware/sbnci/pull/7/commits/eb28b99689f5ae2d0284afce23a8361588ee8b5b)